File: official/docs/java/shipments/RetrieveShipmentList.java
import com.easypost.EasyPost;
import java.util.*;

public class RetrieveShipmentList {
    public static void main(String[] args) {
        EasyPost.apiKey = System.getenv("EASYPOST_API_KEY");
        
        Map<String, Object> listParams = new HashMap<>();
        listParams.put("page_size", 5);

        ShipmentCollection shipments = Shipment.all(listParams);

        System.out.println(shipments);
    }
}

File: official/docs/csharp/shipments/RetrieveShipmentList.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Xunit;
using Xunit.Sdk;
using Newtonsoft.Json;
using EasyPost;

namespace EasyPostExamples;

public class Examples
{
    [Fact]
    public async Task RetrieveAllShipments()
    {
        string apiKey = Environment.GetEnvironmentVariable("EASYPOST_API_KEY")!;

        EasyPost.ClientManager.SetCurrent(apiKey);

        Dictionary<string, object> listParams = new Dictionary<string, object>()
        {
            {
                "page_size", 5
            }
        };

        ShipmentCollection shipmentCollection = await Shipment.All(listParams);

        new TestOutputHelper().WriteLine(JsonConvert.SerializeObject(shipmentCollection, Formatting.Indented));
    }
}
